File: src/routes/token_routes.py
from fastapi import APIRouter, HTTPException
from services.token_service import get_top_pumping_tokens, analyze_token
from models.token import TokenAnalysis

router = APIRouter()

@router.get("/top-pumping")
async def get_pumping_tokens():
    try:
        return await get_top_pumping_tokens()
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.post("/analyze")
async def analyze_token_route(contract_address: str) -> TokenAnalysis:
    try:
        return await analyze_token(contract_address)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: src/routes/tokenRoutes.js
import express from 'express';
import { getTopPumpingTokens, analyzeToken } from '../services/tokenService.js';

const router = express.Router();

router.get('/top-pumping', async (req, res) => {
  try {
    const tokens = await getTopPumpingTokens();
    res.json(tokens);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

router.post('/analyze', async (req, res) => {
  try {
    const { contractAddress } = req.body;
    const analysis = await analyzeToken(contractAddress);
    res.json(analysis);
  } catch (error) {
    res.status(500).json({ error: error.message });
  }
});

export const tokenRoutes = router;